bugbug/repository.py

```diff
@@ -411,131 +352,99 @@ def download_commits(repo_dir, date_from):
         path: "::".join(component) for path, component in path_to_component.items()
     }
 
-    global components_touched_prev
-    global components_touched_prev_90_days
+    first_pushdate = commits[0].pushdate
 
-    global files_touched_prev
-    global files_touched_prev_90_days
+    experiences = defaultdict(lambda: defaultdict(lambda: defaultdict(int)))
+    complex_experiences = defaultdict(lambda: defaultdict(lambda: defaultdict(set)))
 
-    global directories_touched_prev
-    global directories_touched_prev_90_days
+    def update_experiences(experience_type, day, items):
+        for item in items:
+            exp = experiences[day][experience_type][item]
 
-    components_touched = defaultdict(int)
-    files_touched = defaultdict(int)
-    directories_touched = defaultdict(int)
-    prev_commits_90_days = []
-    for commit in commits:
-        components = set(
-            path_to_component[path]
-            for path in commit.files
-            if path in path_to_component
-        )
+            experiences_by_commit["total"][experience_type][commit.node] += exp
+            experiences_by_commit["90_days"][experience_type][commit.node] += (
+                exp - experiences[day - 90][experience_type][item]
+            )
 
-        for component in components:
-            components_touched_prev[commit.node] += components_touched[component]
+            # We don't want to consider backed out commits when calculating experiences.
+            if not commit.backedoutby:
```

Comment:
I'd say that even backed out patches contribute to your experience and maybe your learn more in such a case...
Maybe we could create a special experience for backed out patches: it could help to measure "seniority" (even if I'm not sure of the definition of this word).

Reply:
I mainly didn't want to overcount them, as most of the cases a patch is backed out and then relanded after a fix.
I'll file a follow-up issue to have a special experience for backed out patches.

Reply:
Filed #373.